### ISSUE TITLE
[Intern Progress] Emmanuel Adeitan - Ersilia Model Contribution

### ISSUE THREAD

**Issue by Emzykings** [ersilia-os/outreachy-contributions#27]
# 🌟 Intern Progress Tracker  
Welcome! Use this issue to track your progress as you work through the AI/ML modeling exercise. Follow each step, check the boxes when completed, and update your issue as you go.

# ▶️ **Entry tasks**
- [x] Motivation letter posted in this issue
- [x] Installed the Ersilia Model Hub and successfully run models!
- [x] Recorded system settigns in this issue
- [x] Explained completed tests of Ersilia in the issue

---

##  **1. Download a Dataset of Interest**
- [x] Created a fork of this repository to work on
- [x] Selected a dataset from Therapeutics Data Commons (TDC)
- [x] Verified that the dataset is a classification problem*
- [x] Read and understood the dataset’s background and endpoint  
- [x] Assessed computational feasibility (size, features, etc.)  
- [x] Downloaded the dataset using Python  
- [x] Saved data in the `/data` folder  
- [x] Started writing documentation in the README (Installation, dataset details, how to run code)  

---

##  **2. Featurise the Data**
- [ ] Browsed Ersilia Model Hub for "Representation" models  
- [ ] Selected an appropriate featuriser and justified my choice  
- [ ] Implemented featurisation in a notebook or script
- [ ] Ensured the process is reproducible 
- [ ] Updated the README with featurisation instructions  

---

##  **3. Build an ML Model**
- [ ] Selected an ML framework
- [ ] Trained the model on featurised data  
- [ ] Evaluated the model (accuracy, precision, recall, etc.)  
- [ ] Created graphs using Matplotlib for better visualization  
- [ ] Ensured all steps are reproducible  
- [ ] Documented the model-building process in README

---

##  **4. Prepare Code for Review**
- [ ] Double-checked that documentation is clear and complete  
- [ ] Evaluated if my model performance is satisfactory 
- [ ] Identified possible improvements  
- [ ] Updated this issue to request a review 

---

##  **5. Stretch Tasks (Optional)**
- [ ] Tried a different **featuriser**  
- [ ] Tested other **ML architectures**  
- [ ] Applied the model to **public datasets (e.g., ChEMBL)**  
- [ ] Any other improvements? (Describe below)  

---

## 📌 **Requesting Review**
Once you’ve completed **Steps 1-4**, request a review by mentioning a mentor:  
👤 *@GemmaTuron* Please review my work!  

---

### 📝 **Asking questions**
Use this issue to update your progress, explain where you are stuck or ask for review!

Thank you for contributing! 🎉

**Comment by Emzykings** [comment]
Hello @GemmaTuron ,

My name is Emmanuel Adeitan, a first-class Business Analytics graduate with specialization in AI and Automation Form Nexford University Washington DC, USA. I am a data scientist and machine learning engineer with an intensive passion of using AI to solve real-world challenges in healthcare. My journey in AI began with a strong curiosity of how data-driven solutions could improve life.

Here are some highlight of projects i have worked on one was developing an AI-powered vaccine prediction and supply chain optimization system during E4health hackathon. This experience strengthened my belief in AI's ability to revolutionize biomedical research, ensuring that life-saving remedies reach those who need them most. I have also worked on data-powered solutions in Fintech and e-commerce, to handle large-scale data pipelines and to customize the workflows, and recently my achievement in Datafestafrica 2024 Datathon, where I worked on a significant problem for the Oginigba Comprehensive Secondary School, here is the detail of the [Project](https://docs.google.com/document/d/1T-ww_8yO1VpVhwoCL7EaOarNVxXnHlDFek9bPSLEZcM/edit?tab=t.0)
Beyond my technical expertise, I am emotional about open-source contribution and sharing knowledge. I have mentored aspiring data scientists, conducted training sessions, and actively contributed to developer communities. I believe that AI should be accessible to all.

I am especially excited about contributing to the Ersilia’s Model Hub, as it a perfect intersection of my skills and aspirations. Also the ML for drug discovery and infectious disease research is mission I'm fully aligned with, I see it as an opportunity to collaborate with magnificent minds, to refine my expertise in biomedical AI, and contribute to impressive solutions that solve global health challenges.
By the end of this internship, I wish to enhance my skills in AI research and deployment, implement my learning for real-world healthcare projects, and empower local communities with AI-managed healthcare innovations, especially in maternal and child health. Additionally, I am planning to document my journey through blogs and community attachments to motivate others to detect AI for social good.

I am actually excited about the possibility of working with Ersilia’s team and fellow outreachy applicants. Ready to make meaningful contribution!

Best regards,
Emmanuel Adeitan.

**Comment by Emzykings** [comment]
🛠 Installation & Setup of Packages
To set up the environment, I closely followed the official documentation and successfully completed the installation process.

Steps Taken:
✅ Installed Ubuntu on Windows
✅ Set up Python and Miniconda
✅ Installed Git, GitHub CLI, and Git-LFS
✅ Installed Docker
✅ Installed Ersilia

After completing the setup, I confirmed that Ersilia was properly installed by running the following commands:

bash
Copy code
ersilia --help  
ersilia catalog  
Once Ersilia was recognized in the command line, I tested a basic model as a functionality check:

bash
Copy code
ersilia -v fetch eos3b5e  
ersilia serve eos3b5e  
ersilia -v run -i "CCCC"  
The test produced the expected output, confirming that Ersilia was running correctly.

Ersilia Model Testing & Analysis
To explore Ersilia’s capabilities, I utilized three models to analyze ibuprofen through the CLI. These models included:

1️⃣ eos3b5e – Molecular weight calculator
2️⃣ eos2r5a – Retrosynthetic accessibility predictor
3️⃣ eos9ei3 – Synthetic accessibility score predictor

1️⃣ Molecular Weight Model (eos3b5e)
First, I retrieved and started the model:

bash
Copy code
ersilia fetch eos3b5e  
ersilia serve eos3b5e  
Next, I applied it to ibuprofen using its SMILES representation:

bash
Copy code
ersilia run -i "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"  
✅ Output: The calculated molecular weight of ibuprofen was 206.285 g/mol (rounded to three decimal places).

2️⃣ Retrosynthetic Accessibility Model (eos2r5a)
This model estimates how easily a molecule can be broken down into simpler components during synthesis. The output also included metadata such as key properties, input data, and a textual description.

🔹 Result: The model assigned ibuprofen a high value, indicating that breaking it down into its basic building blocks is quite challenging.

3️⃣ Synthetic Accessibility Score (eos9ei3)
The sa-score model assesses how easily a given molecule can be synthesized in a lab. Similar to the previous model, the results included metadata providing additional insights.

Exploring Ersilia's Model Catalog
To gather more details about each model, I used:

bash
Copy code
ersilia info  
Additionally, I reviewed the list of models I had interacted with by running:

bash
Copy code
ersilia catalog

**Comment by Emzykings** [comment]
### Installing and successfully running  Ersilia Model Hub models!
To set up the environment, I closely followed the official [documentation](https://ersilia.gitbook.io/ersilia-book/ersilia-model-hub/installation) and successfully completed the installation process.

After completing the setup, I confirmed that Ersilia was properly installed by running the following commands:
`ersilia --help  
ersilia catalog`
Once Ersilia was recognized in the command line, I tested a basic model as a functionality check:
`ersilia -v fetch eos3b5e  
ersilia serve eos3b5e  
ersilia -v run -i "CCCC"  
`
The test produced the expected output, confirming that Ersilia was running correctly.

### Ersilia Model Testing & Analysis
To explore Ersilia’s capabilities, I utilized three models to analyze ibuprofen through the CLI. These models included:
1️⃣ eos3b5e – Molecular weight calculator
2️⃣ eos2r5a – Retrosynthetic accessibility predictor
3️⃣ eos9ei3 – Synthetic accessibility score predictor

1️⃣ Molecular Weight Model (eos3b5e)
First, I retrieved and started the model:
`ersilia fetch eos3b5e  
ersilia serve eos3b5e `
Next, I applied it to ibuprofen using its SMILES representation:
`ersilia run -i "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O" `
✅ Output: The calculated molecular weight of ibuprofen was 206.285 g/mol (rounded to three decimal places).

![Image](https://github.com/user-attachments/assets/39dce5d1-7818-4e14-b219-6881069d3d2f)

2️⃣ Retrosynthetic Accessibility Model (eos2r5a)
This model estimates how easily a molecule can be broken down into simpler components during synthesis. The output also included metadata such as key properties, input data, and a textual description.

🔹 Result: The model assigned ibuprofen a high value, indicating that breaking it down into its basic building blocks is quite challenging.

![Image](https://github.com/user-attachments/assets/c4dc3318-206e-41e7-a3f7-f593606a4762)

3️⃣ Synthetic Accessibility Score (eos9ei3)
The sa-score model assesses how easily a given molecule can be synthesized in a lab. Similar to the previous model, the results included metadata providing additional insights.

![Image](https://github.com/user-attachments/assets/e1853f77-e7a2-4d04-a63c-3a8641a5d889)

### Exploring Ersilia's Model Catalog
To gather more details about each model, I used:
`ersilia info`
Additionally, I reviewed the list of models I had interacted with by running:
`ersilia catalog`

![Image](https://github.com/user-attachments/assets/0ec60cc9-4453-415a-9fe4-cea04ec9f2e0)

![Image](https://github.com/user-attachments/assets/b00a5b38-00fa-461a-bf70-641052d885cb)

**Comment by Emzykings** [comment]
### system settigns
Operating system: Ubuntu 24.04.2 LTS (WSL)
Python: Python 3.12.9
Docker: Docker version 26.1.3, build 26.1.3-0ubuntu1~24.04.1
Conda: conda 25.1.1

![Image](https://github.com/user-attachments/assets/c9e97a11-dd7a-4c4a-9c01-ad847ee2c97a)

**Comment by Emzykings** [comment]
@GemmaTuron, kindly review my progress!

**Comment by Emzykings** [comment]
@GemmaTuron kindly review my progress!

**Comment by GemmaTuron** [comment]
Hello @Emzykings 

Welcome to Ersilia and thanks for the effort in the motivation letter. Please go ahead and submit your contribution to outreachy to start tackling weeks2-3 tasks

**Comment by Emzykings** [comment]
✅ Week 2 Progress Update – Dataset Selection, Preparation & Documentation
Contributor: Emmanuel Adeitan([@Emzyking](https://github.com/Emzykings))
Branch: emmanuel-week2-task , which would be merged to main @GemmaTuron i have created a PR

🧠 Summary of Work Completed
I have completed all checklist items for Week 2: Downloading and Understanding my Dataset of Interest from the [Outreachy Contribution Guide](https://github.com/ersilia-os/outreachy-contributions). My goal was to select an appropriate dataset from the Therapeutics Data Commons (TDC), understand its scientific background, prepare it for modeling, and fully document the process.

📊 Dataset Selection & Decision Making
Chosen Dataset: ames from the admet_group of the [Therapeutics Data Commons](https://tdcommons.ai/)

Task Type: Binary classification (Predicting whether a compound is mutagenic)

Target variable: Y → 0 = non-mutagenic, 1 = mutagenic

Why this dataset?

✅ Classification task (as recommended)

✅ Well-studied in drug discovery and ADMET profiling

✅ Small size (~7,000 rows), ideal for local compute

✅ Raw SMILES strings are suitable for Ersilia featurization

✅ Easy to interpret and explain (toxicity prediction)

🔧 Data Processing Details
Used PyTDC (v0.3.5) to download the dataset via:
```
from tdc.benchmark_group import admet_group
group = admet_group()
benchmark = group.get("ames")
```
The dataset came split into:
train_val: combined train/val set
test: holdout set

Applied a stratified split on train_val using scikit-learn to separate:
train: 70%
val: 15%
test: 15% (pre-split)

✅ All files were saved into the /data folder:
ames_train.csv
ames_val.csv
ames_test.csv

✅ Preview of data (Drug, SMILES, Y) printed successfully.

📁 Repo & Script Summary
Script used: notebooks/1_download_dataset.py

Updated and committed a detailed, professional README.md with:

Dataset explanation
Justification for selection
Full installation and execution instructions
File structure

📝 Tasks Marked Completed
 Created a fork and worked on a separate branch
 Selected and explored a TDC dataset
 Verified classification problem type
 Read and understood dataset’s background
 Assessed compute requirements
 Downloaded and saved data via Python
 Stratified train/val/test split applied
 Saved data to /data folder
 Wrote detailed documentation in README.md
 Pushed changes and created a branch for review

🚧 Next Step (Week 3 Preview)
Install Ersilia CLI
Search for a molecular featurization model (representation-based)
Apply to the SMILES column in ames_train.csv, etc.
Document the rationale for chosen featurizer
Proceed to training and evaluating ML models (XGBoost / FLAML / scikit-learn)

**Comment by Emzykings** [comment]
@GemmaTuron  kindly review my progress Thanks!

**Comment by GemmaTuron** [comment]
Hi @Emzykings 

Thanks, I currently do not see the repository in your fork updated, maybe I am looking at the wrong fork? Can you send the right link? And couple questions:
- Is TDC not providing the train and validation splits prepared already?
- Please use the main README to document the project so to have everything in a single page

**Comment by miquelduranfrigola** [comment]
Auto-closing this Outreachy intern-progress tracker as completed — the contribution round has ended. Reopen if follow-up is needed.
